File: leetcode/src/main/java/com/arctos/mochabear/leetcode/model/LinkedListBuilder.java
package com.arctos.mochabear.leetcode.model;

public class LinkedListBuilder {

    public ListNode buildLinkedList(int[] nums) {
        if (nums.length == 0) return null;

        ListNode head = new ListNode(nums[0]);
        ListNode current = head;
        for (int i = 1; i < nums.length; i++) {
            current.next = new ListNode(nums[i]);
            current = current.next;
        }
        return head;
    }

}

File: leetcode/src/main/kotlin/com/arctos/mochabear/leetcode/model/LinkedListBuilder.kt
package com.arctos.mochabear.leetcode.model

class LinkedListBuilder {

    fun buildLinkedList(nums: IntArray): ListNode? {
        if (nums.isEmpty()) return null

        val head = ListNode(nums[0])
        var current: ListNode = head
        for (i in 1 until nums.size) {
            current.next = ListNode(nums[i])
            current = current.next!!
        }
        return head
    }
}